tests/dag_processing/test_manager.py

```diff
@@ -438,17 +439,23 @@ def test_add_new_file_to_parsing_queue(
         )
 
     @conf_vars({("scheduler", "file_parsing_sort_mode"): "modified_time"})
+    @mock.patch("airflow.settings.TIMEZONE", timezone.utc)
     @mock.patch("zipfile.is_zipfile", return_value=True)
     @mock.patch("airflow.utils.file.might_contain_dag", return_value=True)
     @mock.patch("airflow.utils.file.find_path_from_directory", return_value=True)
     @mock.patch("airflow.utils.file.os.path.isfile", return_value=True)
     @mock.patch("airflow.utils.file.os.path.getmtime")
     def test_recently_modified_file_is_parsed_with_mtime_mode(
-        self, mock_getmtime, mock_isfile, mock_find_path, mock_might_contain_dag, mock_zipfile
+        self, mock_getmtime, mock_isfile, mock_find_path, mock_might_contain_dag, mock_zipfile, monkeypatch
     ):
         """
         Test recently updated files are processed even if min_file_process_interval is not reached
         """
+        # ensure platform timezone does not affect the modified time value
+        monkeypatch.setenv("TZ", "Europe/Paris")
+        # propagate new timezone to C routines
+        time.tzset()
```

Comment:
Better to do this in a pytest fixture; as this is currently implemented, timezone would not be reset if the test fails (and does not reach the end of the function).

It’s probably also a good idea to add a `skipif` since `tzset` is not universally available.

Reply:
great idea. I've added a new fixture and applied it to all _mtime_ tests.

Instead of adding a `skipif` mark, I just call `tzset` if it exists. According the the Windows documentation of `localtime_s` [0] (which is used internally by `fromtimestamp` on Windows if no timezone is set)

> localtime_s corrects for the local time zone if the user first sets the global environment variable TZ. When TZ is set, three other environment variables (_timezone, _daylight, and _tzname) are automatically set as well. 

[0] https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/localtime-s-localtime32-s-localtime64-s?view=msvc-170